modo/helpers.py

```diff
@@ -291,14 +301,48 @@ def bytesio_to_iterator(
         temp_file.seek(0)
 
         # Open the temporary file as a pysam.AlignmentFile/VarianFile object
-        if file_format == "CRAM":
-            pysamFile = AlignmentFile(
-                temp_file.name, "rc", reference_filename=reference_filename
-            )
-        elif file_format in ("VCF", "BCF"):
-            # write_mode = r if fileformat=="VCF" else "rb"
-            pysamFile = VariantFile(temp_file.name, "rb")
-        with pysamFile as in_file:
+        pysam_file = file_to_pysam_object(
+            path=temp_file.name,
+            fileformat=file_format,
+            reference_filename=reference_filename,
+        )
+
+        with pysam_file as in_file:
             # Iterate over the alignments in the file
             for line in in_file:
                 yield line
+
+
+def iter_to_file(
+    gen_iter: Iterator[AlignedSegment | VariantRecord],
+    infile,  # [AlignmentFile | VariantFile]
+    output_filename: str,
+    reference_filename: Optional[str] = None,
+):
+    out_fileformat = get_fileformat(output_filename)
+    if out_fileformat in ("CRAM", "BAM", "SAM"):
+        write_mode = (
+            "wc"
+            if out_fileformat == "CRAM"
+            else ("wb" if out_fileformat == "BAM" else "w")
+        )
+        output = AlignmentFile(
+            output_filename,
+            mode=write_mode,
+            template=infile,
+            reference_filename=reference_filename,
+        )
+    elif out_fileformat in ("VCF", "BCF"):
+        write_mode = "w" if out_fileformat == "VCF" else "wb"
+        output = VariantFile(
+            output_filename, mode=write_mode, header=infile.header
+        )
+    else:
+        raise ValueError(
+            "Unsupported output file type. Supported files: .cram, .bam, .sam, .vcf, .vcf.gz, .bcf."
+        )
+
+    for read in gen_iter:
+        output.write(read)
+    output.close()
+    return None
```

Comment:
```suggestion
```
**suggestion:** if this function never returns anything, we may as well drop the return statement altogether

Reply:
Done, see https://github.com/sdsc-ordes/modo-api/commit/6633176f4e69390aa7afe4cd925b791a1a94bc9a

Reply:
Done both.

Remove return None - see https://github.com/sdsc-ordes/modo-api/commit/2f5ee25f4dce980886b32f5740d91cb9a2cbfd99

Update modo-schema via poetry - see https://github.com/sdsc-ordes/modo-api/commit/d1ae6d8bb7438a35d1aebcd8f07a803ae511a03f